go-controller/pkg/node/healthcheck.go

```diff
@@ -95,6 +96,22 @@ func countLocalEndpoints(ep *kapi.Endpoints, nodeName string) int {
 	return num
 }
 
+func countHostNeworkEndpoints(ep *kapi.Endpoints, nodeAddresses *sets.String) bool {
+	out := false
+	for i := range ep.Subsets {
+		ss := &ep.Subsets[i]
+		for i := range ss.Addresses {
+			addr := &ss.Addresses[i]
+			if nodeAddresses.Has(addr.IP) {
+				out = true
+				break
```

Comment:
just return `true` directly here, if just want to check if there is at least one hostNetwork endpoint
This `break` is in a nested loop, it looks that you'll keep processing more addresses
I think that @trozet already has a function for this